datacube/utils/geometry/tools.py

```diff
@@ -446,20 +446,24 @@ def box_overlap(src_shape, dst_shape, ST, tol):
                       direction is: Xsrc = ST*Xdst
     :param       tol: Sub-pixel translation tolerance that's scaled by resolution.
     """
-    from datacube.utils.math import is_almost_int
+    from datacube.utils.math import maybe_int
 
     (sx, _, tx,
      _, sy, ty,
      *_) = ST
 
-    def _round_to_near_int(x, tol):
-        tol = min(0.5, tol)
-        if is_almost_int(x, tol):
-            x = round(x)
-        return x
+    def snap_scale(s, tol=1e-6):
```

Comment:
Can we move this out of this function and add tests for it?
Remember that scale can be negative,  this case should be handled properly.